book/1-pre-project-phase/key_areas_in_data.md

```diff
@@ -0,0 +1,75 @@
+
+![Corrplot](../assets/images/charts/corrplot_chart.png)  
+
+At times, you will face a situation in which sponsor claims that their domain knowledge is accurate in determining the predictive features while our exploratory data analysis (EDA) lends limited support for it. You could attempt some denoising techniques (e.g. outlier removal, averaging/binning) and check if the correlation improves. If it does not improve, this might be due to some confounding effect of a third feature that may influence the relationship between target and feature of interest. 
+
+3. Is the data at the correct granularity?
+
+It is critical to ensure that the training data and production data share the same level of granularity (e.g. hourly, daily, weekly, monthly or yearly) as what your sponsor expects in prediction. So if the sponsor expects the prediction on an hourly basis and the sponsor collects the data daily, then it will not be possible to build such an AI model to produce a granular prediction based on aggregated data. It is also worthy to note that if expected prediction is at a higher or similar granularity as the training or production data, then it is possible to build an AI model. 
+
+![GranularityDiagram](../assets/images/diagrams/granularity_diagram.jpg)  
+
+Besides matching granularity between training and production data, understanding the correct granularity allows you to estimate the volume of data available for training. For example, if the goal is to aggregate a sensor's readings from seconds to hours for prediction, because the per-second readings contain mostly noise, then having "86,400 data points" may sound like a large dataset, but in actual fact, it is insufficient as there are only 24 hourly aggregates for you to work with.
+
+4. Is the training data representative of the production data?
+
+It is critical that training data is representative of the production data to ensure the trained model can attain decent prediction performances with following considerations:
+
+- Avoid unintended bias. If the model is trained on training data that is dominated by a specific label (i.e. majority label), then it is very likely that the model is robust in predicting a majority label while it will unlikely predict a minority label. 
+
+- Improve robustness and generalisability. The distribution of training data will be what the model will learn from. Assuming that the distribution between training and production data is highly similar, this would mean that the trained model is able to generalise and make robust predictions in varying contexts that are often present in production data.  
+
+- Mitigate "instant drift" the moment when the model is deployed into production. This can be observed by setting up input metrics to monitor the input distribution in production against the one in training or output metrics to check if the model is unable to confidently predict classes across many data points. This would signal a need for retraining the model. 
```

Comment:
```- Mitigate "instant drift" the moment when the model is deployed into production. This can be observed by setting up input metrics to monitor the input distribution in production against the one in training or output metrics to check if the model is unable to confidently predict classes across many data points. This would signal a need for retraining the model.```
We can omit this point in view of the length.